### PR TITLE
adds metric prefix.

### PR DESCRIPTION
**Pull request by riklopfer**
# What does this PR do?

Adds metric prefix to metrics dict. This is needed for  `metric_for_best_model` to function properly. See https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L1516

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

- maintained examples (not research project or legacy): @sgugger, @patil-suraj

### CODE REVIEW

File: examples/pytorch/question-answering/trainer_qa.py [hunk 1]
```diff
@@ -92,4 +97,9 @@ def predict(self, predict_dataset, predict_examples, ignore_keys=None):
         predictions = self.post_process_function(predict_examples, predict_dataset, output.predictions, "predict")
         metrics = self.compute_metrics(predictions)
 
+        # Prefix all keys with metric_key_prefix + '_'
+        for key in list(metrics.keys()):
+            if not key.startswith(f"{metric_key_prefix}_"):
+                metrics[f"{metric_key_prefix}_{key}"] = metrics.pop(key)
```
Comment (sgugger) [inline examples/pytorch/question-answering/trainer_qa.py:103]: This shouldn't be necessary here, instead pass along the `metric_key_prefix` to the `eval_loop` above.

File: examples/pytorch/question-answering/trainer_qa.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def evaluate(self, eval_dataset=None, eval_examples=None, ignore_keys=None):
             eval_preds = self.post_process_function(eval_examples, eval_dataset, output.predictions)
             metrics = self.compute_metrics(eval_preds)
 
+            # Prefix all keys with metric_key_prefix + '_'
+            for key in list(metrics.keys()):
+                if not key.startswith(f"{metric_key_prefix}_"):
+                    metrics[f"{metric_key_prefix}_{key}"] = metrics.pop(key)
```
Comment (sgugger) [inline examples/pytorch/question-answering/trainer_qa.py:62]: This shouldn't be necessary here, instead pass along the `metric_key_prefix` to the `eval_loop` above.

**Review by sgugger — COMMENTED** [review]
Thanks for tackling this! I think we should do it the same way as in the main Trainer class though.

File: examples/pytorch/question-answering/trainer_qa.py (hunk 1 above)
Comment (riklopfer) [reply examples/pytorch/question-answering/trainer_qa.py:103]: That's where I thought it should go at first too. But, it seems that metrics are disabled _before_ calling `eval_loop`.

Comment (sgugger) [reply examples/pytorch/question-answering/trainer_qa.py:103]: Ah yes, forgot that specificity of the QA Trainer, thanks for reminding me! It's good like this!

**Review by sgugger — APPROVED** [review]

File: examples/pytorch/question-answering/trainer_qa.py (hunk 2 above)
Comment (riklopfer) [reply examples/pytorch/question-answering/trainer_qa.py:62]: Same deal here -- metrics are disabled before `eval_loop`

**Comment by riklopfer** [comment]
@sgugger, the example tests are fixed now, this other failure is a mystery to me. I suspect that failure is caused by another change. Let me know if you think the `run_torch_tests` failure is related to this PR and I'll look into it.

**Comment by sgugger** [comment]
No this failure is independent and currently being investigated, so we can merge this PR safely. Thanks again!

**Pull request merged**